datacube_stats/main.py

```diff
@@ -940,5 +965,168 @@ def __call__(self, product, time, group_by):
+
+
+# def Generate_Polygon_Tasks(index, sources_spec, date_ranges, filter_product, input_region, storage):
+class Generate_Polygon_Tasks(object):
+    """
+    Make stats tasks for a defined spatial region, that doesn't fit into a standard grid.
+    :param index: database index
+    :param input_region: dictionary of query parameters defining the target input region. Usually
+                         x/y spatial boundaries.
+    :return:
+    """
+    def __init__(self, input_region, filter_product, storage):
+        self.input_region = input_region
+        self.filter_product = filter_product
+        self.storage = storage
+        self.tile_index = ()
+        self.filter_time = ""
+        self.feature = {}
+
+    def __call__(self, index, sources_spec, date_ranges):
+        """
+        Generate the required tasks through time and across a feature ID of a polygon.
+
+        feature id is passed through filter_product and is processed after filtering the relevant dates
+        that will both result in only datasets covering the poly.
+
+        :param index: Datacube Index
+        :return:
+        """
+        if self.filter_product['args'].get('feature_id') is None:
+            _LOG.info("feature_id must be provided.  EXITING ...")
+            return
+        with fiona.open(self.input_region['from_file']) as input_region:
+            for feature in input_region:
+                Id = feature['properties']['ID']
+                # capture filename and feature id to get boundary info in execute task
+                if Id in self.filter_product['args']['feature_id']:
+                    self.feature = feature['properties']
+                    self.filter_product['filename'] = self.input_region['from_file']
+                    break
+            if len(self.feature) == 0:
+                _LOG.info("No feature id is found. Please check the ID in config file under filter_product")
+                return
+        make_tile = FilteredTileMaker(index, self.filter_product, self.storage)
+        dc = make_tile(product=None, time=None, group_by=None, filter_time=None, geopoly=None, poly_dates=1)
+        boundary_polygon = geom_from_file(self.filter_product)
+        # Get all dates within the date_range
+        all_dates = list_poly_dates(dc, boundary_polygon, sources_spec, date_ranges)
+        # Test for dry/wet products
+        if self.filter_product.get('method') == 'by_hydrological_months':
+            # get the year id and geometry
+            self.filter_product['year'] = {k: v for k, v in self.feature.items() if "DY" in k.upper()} \
+                 if self.filter_product.get('product') == 'dry' else  \
+                 {k: v for k, v in self.feature.items() if "WY" in k.upper()}
+            years = "_".join(x for x in [v for k, v in self.filter_product['year'].items()])
+            nlon = self.filter_product['args']['name'].upper() + '_' + str(Id)
+            self.tile_index = (nlon, years)
```

Comment:
I also don't think that it makes sense to have a `tile_index` here. Aren't we dealing with polygons?